libs/agno/agno/models/base.py

```diff
@@ -1423,7 +1423,7 @@ async def arun_function_call(
             raise e
 
         function_call_timer.stop()
-        return success, function_call_timer, function_call, result.updated_session_state
+        return success, function_call_timer, function_call, result.updated_session_state, result
```

Comment:
Lets not unpack the updated_session_state then and just return result (in the last part) and handle it downstream